inc/zoo/swar/SWAR.h

```diff
@@ -62,13 +73,53 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
```

Comment:
What is the reason for this `if constexpr`? because the ternary operator `?` would get the compiler to warn on the branch that would not be active?

Reply:
At one point the compiler was failing at the ternary operator, so I moved it over to this.

Reply:
Yes, there is no option.